# HG changeset patch
# User Rik <rik@octave.org>
# Date 1388806212 28800
#      Fri Jan 03 19:30:12 2014 -0800
# Branch stable
# Node ID 8d7e45ea11fa416ee2f9ca6deb5bcb5834c22a47
# Parent  a0abcf377ec5838a6229c2e33054e649425c9740
doc: Add new octave-cli man page (bug #41085).

* octave-cli.1: New man page.

* doc/interpreter/Makefile.am: Add octave-cli.1 to build system.

* mkoctfile.1: Move "SEE ALSO" section to end of man page.

*octave-config.1: Add "SEE ALSO" section to end of man page.

diff --git a/doc/interpreter/Makefile.am b/doc/interpreter/Makefile.am
--- a/doc/interpreter/Makefile.am
+++ b/doc/interpreter/Makefile.am
@@ -27,16 +27,17 @@ TEXINPUTS := "$(PATH_SEPARATOR)$(srcdir)
 export TEXINPUTS
 
 ## Include custom texmf.cnf necessary to run @seealso macro 
 TEXMFCNF := "..$(PATH_SEPARATOR)$(srcdir)/..$(PATH_SEPARATOR)$(TEXMFCNF)$(PATH_SEPARATOR)"
 export TEXMFCNF
 
 dist_man_MANS = \
   mkoctfile.1 \
+  octave-cli.1 \
   octave-config.1 \
   octave.1
 
 ## The following files are included in the manual via the @EXAMPLEFILE macro.
 ## They are dependencies for the documentation.
 ## They must not be distributed from this directory.
 EXAMPLE_FILES = \
   $(top_srcdir)/examples/@FIRfilter/FIRfilter.m \
diff --git a/doc/interpreter/mkoctfile.1 b/doc/interpreter/mkoctfile.1
--- a/doc/interpreter/mkoctfile.1
+++ b/doc/interpreter/mkoctfile.1
@@ -23,17 +23,17 @@
 .ft CW
 .nf
 ..
 .de Ve \" (V)erbatim (e)nd.  Return to regular font and justification
 .ft R
 .fi
 ..
 .\" --------------------------------------------------------------------
-.TH MKOCTFILE 1 "4 February 2011" "GNU Octave"
+.TH MKOCTFILE 1 "3 January 2014" "GNU Octave"
 .SH NAME
 mkoctfile \- Compile dynamic-load modules for GNU Octave
 .SH SYNOPSIS
 \fBmkoctfile\fP [\fIOPTION\fP]... file...
 .SH DESCRIPTION
 \fBmkoctfile\fP is used to compile C, C++, or Fortran source code in
 to a dynamically loadable \fI.oct\fP file for
 .BR octave (1).
@@ -135,16 +135,16 @@ Compile or link file.  Recognized file t
    .f    Fortran source (fixed form)
    .F    Fortran source (fixed form)
    .f90  Fortran source (free form)
    .F90  Fortran source (free form)
    .o    object file
    .a    library file
 .Ve
 .RE
-.SH "SEE ALSO"
-octave (1).
 .SH AUTHOR
 John W. Eaton <jwe@octave.org>
 
 This manual page was contributed by Dirk Eddelbuettel
 <edd@debian.org> for the Debian GNU/Linux distribution but
 may be used by others.
+.SH "SEE ALSO"
+octave (1).
diff --git a/doc/interpreter/octave-cli.1 b/doc/interpreter/octave-cli.1
new file mode 100644
--- /dev/null
+++ b/doc/interpreter/octave-cli.1
@@ -0,0 +1,86 @@
+.\" Copyright (C) 1996-2014 John W. Eaton
+.\"
+.\" This file is part of Octave.
+.\"
+.\" Octave is free software; you can redistribute it and/or modify it
+.\" under the terms of the GNU General Public License as published by the
+.\" Free Software Foundation; either version 3 of the License, or (at
+.\" your option) any later version.
+.\"
+.\" Octave is distributed in the hope that it will be useful, but WITHOUT
+.\" ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+.\" FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+.\" for more details.
+.\"
+.\" You should have received a copy of the GNU General Public License
+.\" along with Octave; see the file COPYING.  If not, see
+.\" <http://www.gnu.org/licenses/>.
+.\" --------------------------------------------------------------------
+.de Vb \" (V)erbatim (b)egin.  Use fixed width font and no justification
+.ft CW
+.nf
+..
+.de Ve \" (V)erbatim (e)nd.  Return to regular font and justification
+.ft R
+.fi
+..
+.\" --------------------------------------------------------------------
+.TH OCTAVE-CLI 1 "03 January 2014" "GNU Octave"
+.SH NAME
+octave-cli \- A high-level interactive language for numerical computations.
+.SH SYNOPSIS
+\fBoctave-cli\fP [\fIoptions\fP]... [\fIfile\fP]
+.SH DESCRIPTION
+Octave is a high-level language, primarily intended for numerical
+computations.  It provides a convenient command line interface for
+solving linear and nonlinear problems numerically.  \fBoctave-cli\fP
+has been compiled without any GUI support (Qt) which makes it smaller
+than the default \fBoctave\fP executable, but also limits it to
+providing just the command line interface (CLI).
+.SH OPTIONS
+The complete set of command-line options for \fBoctave-cli\fP is available
+by running the following command from the shell.
+.Vb
+
+    octave-cli \-\-help
+.Ve
+.SH DOCUMENTATION
+The primary documentation for Octave is written using Texinfo, the GNU
+documentation system, which allows the same source files to be used to
+produce online and printed versions of the manual.
+.PP
+You can read the online copy of the Octave documentation by issuing
+the following command from within Octave.
+.Vb
+
+    octave:1> doc
+
+.Ve
+The Info files may also be read with a stand-alone program such as
+\fBinfo\fP or \fBxinfo\fP.  HTML, Postscript, or PDF versions of the
+documentation are installed on many systems as well.
+.SH BUGS
+The Octave project maintains a bug tracker at http://bugs.octave.org.
+Before submitting a new item please read the instructions at
+http://www.octave.org/bugs.html on how to submit a useful report.
+.SH FILES
+Upon startup Octave looks for four initialization files.  Each file
+may contain any number of valid Octave commands.
+.TP
+\fIoctave-home\fP/share/octave/site/m/startup/octaverc
+Site-wide initialization file which changes options for all users.
+\fIoctave-home\fP is the directory where Octave was installed such as
+\fB/usr/local\fP.
+.TP
+\fIoctave-home\fP/share/octave/\fIversion\fP/m/startup/octaverc
+Site-wide initialization file for Octave version \fIversion\fP.
+.TP
+~/.octaverc
+User's personal initialization file.
+.TP
+\&\.octaverc
+Project-specific initialization file located in the current directory.
+.SH AUTHOR
+John W. Eaton <jwe@octave.org>
+.SH SEE ALSO
+octave (1)
diff --git a/doc/interpreter/octave-config.1 b/doc/interpreter/octave-config.1
--- a/doc/interpreter/octave-config.1
+++ b/doc/interpreter/octave-config.1
@@ -23,17 +23,17 @@
 .ft CW
 .nf
 ..
 .de Ve \" (V)erbatim (e)nd.  Return to regular font and justification
 .ft R
 .fi
 ..
 .\" --------------------------------------------------------------------
-.TH OCTAVE-CONFIG 1 "4 February 2011" "GNU Octave"
+.TH OCTAVE-CONFIG 1 "3 January 2014" "GNU Octave"
 .SH NAME
 octave-config - GNU Octave component and library information retrieval
 .SH SYNOPSIS
 \fBoctave-config\fP [\fIOPTION\fP]...
 .SH DESCRIPTION
 .B octave-config
 is a tool to obtain directory information for \fB.oct\fP and \fB.m\fP files for
 .BR octave (1).
@@ -76,8 +76,10 @@ Display the version number of
 .TP
 \fB\-h\fP, \fB\-?\fP, \fB\-\-help\fP
 Print help message.
 .SH AUTHOR
 John W. Eaton <jwe@octave.org>
 
 This manual page was contributed by Dirk Eddelbuettel <edd@debian.org> 
 for the Debian GNU/Linux distribution but may be used by others.
+.SH "SEE ALSO"
+octave (1).
